state-chain/custom-rpc/src/lib.rs

```diff
@@ -1121,6 +1141,28 @@ where
 		)
 	}
 
+	fn cf_subscribe_scheduled_swaps(
+		&self,
+		sink: SubscriptionSink,
+		base_asset: Asset,
+		quote_asset: Asset,
+	) -> Result<(), SubscriptionEmptyError> {
+		self.new_subscription(
+			false, /* only_on_changes */
+			true,  /* end_on_error */
+			sink,
+			move |api, hash| {
+				let swaps = api
+					.cf_scheduled_swaps(hash, base_asset, quote_asset)?
```

Comment:
Also then you could just delete `SwapLegInfo` and inline it into The ScheduledSwap structure, i.e. avoiding the `serde(flatten)`.

Reply:
I changed `cf_scheduled_swaps` to return `Vec<ScheduledSwap>` now, and simplified the code a bit.

However, I'm not sure merging the two structs is obviously justified. The reason I have separate structs is bacause `get_scheduled_swap_legs` doesn't deal with block numbers, so we'd need to pass the block number through that function just so it can construct `ScheduledSwap` on our behalf, which doesn't seem particularly nice. I would aslo need to add a generic parameter (for the block number) to `ScheduledSwap` to make it work. I don't think having this extra struct is much worse than the alternative. I can make this change if you have a strong preference though.

Reply:
You should just inline get_scheduled_swap_legs, the fact that it causes you to need two struct's here (that aren't used separate other than in this one function), imo is clear indication the split isn't a good idea.

Reply:
The advantage of having a separate `get_scheduled_swap_legs` is that we can write unit tests against it. It is possible to re-write tests with it inlined, but that wouldn't be as nice. I don't think having a separate struct for what the RPC call returns is such a bad thing, particularly since we wrap the internal type (with `serde(flatten)`), and so we don't even have any duplication.